packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -25,18 +25,18 @@ export default function buildConfigChain(opts: Object = {}) {
   const filename = opts.filename;
   const builder = new ConfigChainBuilder();
 
-  // resolve all .babelrc files
-  if (opts.babelrc !== false) {
-    builder.findConfigs(filename);
-  }
-
   builder.mergeConfig({
```

Comment:
By swapping the order of these, we'll be able to bail out based on `ignore` and `only` match failure, before having to search for the `.babelrc` which will be a big performance win.